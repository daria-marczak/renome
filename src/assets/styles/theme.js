export const theme = {
  primary: '#985e5b',
  black: '#000',
  white: '#fff',
  lightGray: '#dadada',
  gray: '#545459',
  font: {
    family: {
      lato: 'Lato, sans-serif',
      montserrat: 'Montserrat, sans-serif',
      playfair: 'Playfair Display, serif',
    },
    size: {
      header: '38px',
      paragraph: '14px',
      jumbotron: '22px',
      paragraphDescription: '11px',
      carouselDescription: '16px',
    },
  },
};
