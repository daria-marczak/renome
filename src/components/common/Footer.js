import React from 'react';
import styled from 'styled-components';

import Logo from '../../assets/images/logo.png';
import Facebook from '../../assets/icons/fb.png';
import Google from '../../assets/icons/gplus.png';
import Twitter from '../../assets/icons/twitter.png';

const StyledFooter = styled.footer`
  background-color: ${({ theme }) => theme.lightGray};
  text-align: center;
  padding: 20px 0;

  @media (min-width: 1200px) {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
  }
`;

const StyledParagraph = styled.p`
  font-weight: 700;
  text-transform: uppercase;
  font-size: ${({ theme }) => theme.font.size.paragraph};
`;

const StyledContainer = styled.div`
  display: flex;
  justify-content: center;
  margin-bottom: 20px;

  @media (min-width: 1200px) {
    margin-bottom: 0;
    margin-left: 10px;
  }
`;

const StyledIcon = styled.button`
  height: 30px;
  width: 30px;
  border-radius: 100%;
  display: flex;
  justify-content: center;
  background-color: transparent;
  align-items: center;
  border: 2px solid ${({ theme }) => theme.black};
  margin-right: 10px;
`;

const StyledFollowColumn = styled.div`
  @media (min-width: 1200px) {
    display: flex;
    align-items: center;
    order: 3;
    justify-self: center;
  }
`;

const StyledColumn = styled.div`
  @media (min-width: 1200px) {
    align-self: center;
  }
`;

const Footer = () => (
  <StyledFooter>
    <StyledFollowColumn>
      <StyledParagraph>Follow us:</StyledParagraph>
      <StyledContainer>
        <StyledIcon>
          <img src={Facebook} alt="facebook" />
        </StyledIcon>
        <StyledIcon>
          <img src={Google} alt="google plus" />
        </StyledIcon>
        <StyledIcon>
          <img src={Twitter} alt="twitter" />
        </StyledIcon>
      </StyledContainer>
    </StyledFollowColumn>
    <StyledColumn>
      <img src={Logo} alt="logo" />
    </StyledColumn>
    <StyledColumn>
      <StyledParagraph>Copyright 2015 renome by estetiq</StyledParagraph>
    </StyledColumn>
  </StyledFooter>
);

export default Footer;
