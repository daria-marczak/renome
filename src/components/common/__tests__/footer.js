import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Footer from '../Footer';
import theme from '../../../assets/styles/theme';

describe('Footer', () => {
  it('renders correctly', () => {
    const tree = new ShallowRenderer(<Footer theme={theme} />);
    expect(tree).toMatchSnapshot();
  });
});
