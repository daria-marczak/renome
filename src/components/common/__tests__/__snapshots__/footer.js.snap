// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
ReactShallowRenderer {
  "_context": null,
  "_didScheduleRenderPhaseUpdate": false,
  "_dispatcher": Object {
    "readContext": [Function],
    "useCallback": [Function],
    "useContext": [Function],
    "useDebugValue": [Function],
    "useEffect": [Function],
    "useImperativeHandle": [Function],
    "useLayoutEffect": [Function],
    "useMemo": [Function],
    "useReducer": [Function],
    "useRef": [Function],
    "useState": [Function],
  },
  "_element": null,
  "_firstWorkInProgressHook": null,
  "_forcedUpdate": false,
  "_instance": null,
  "_isReRender": false,
  "_newState": null,
  "_numberOfReRenders": 0,
  "_renderPhaseUpdates": null,
  "_rendered": null,
  "_rendering": false,
  "_updater": Updater {
    "_callbacks": Array [],
    "_renderer": [Circular],
  },
  "_workInProgressHook": null,
}
`;
